pkg/drivers/powerflex.go

```diff
@@ -70,96 +70,79 @@ func PrecheckPowerFlex(ctx context.Context, cr *csmv1.ContainerStorageModule, op
 		return fmt.Errorf("%s %s not supported", csmv1.PowerFlexName, cr.Spec.Driver.ConfigVersion)
 	}
 
-	mdmVar, err := GetMDMFromSecret(ctx, cr, ct)
+	// Check if MDM is set in the secret
+	_, err := GetMDMFromSecret(ctx, cr, ct)
 	if err != nil {
 		return err
 	}
-	var newmdm corev1.EnvVar
+
+	return nil
+}
+
+func SetSDCinitContainers(ctx context.Context, cr csmv1.ContainerStorageModule, ct client.Client) (csmv1.ContainerStorageModule, error) {
+	mdmVar, _ := GetMDMFromSecret(ctx, &cr, ct)
+
+	// Check if SDC is enabled
 	sdcEnabled := true
-	for _, env := range cr.Spec.Driver.Node.Envs {
-		if env.Name == "X_CSI_SDC_ENABLED" && env.Value == "false" {
-			sdcEnabled = false
+	if cr.Spec.Driver.Node != nil {
+		for _, env := range cr.Spec.Driver.Node.Envs {
+			if env.Name == "X_CSI_SDC_ENABLED" && env.Value == "false" {
+				sdcEnabled = false
+				break
+			}
 		}
 	}
 
-	newInitContainers := make([]csmv1.ContainerTemplate, 0)
+	// Update init containers
+	var newInitContainers []csmv1.ContainerTemplate
 	for _, initcontainer := range cr.Spec.Driver.InitContainers {
-		if initcontainer.Name != "sdc" {
-			newInitContainers = append(newInitContainers, initcontainer)
-		} else if initcontainer.Name == "sdc" && sdcEnabled {
-			k := 0
-			initenv := initcontainer.Envs
-			for c, env := range initenv {
+		if initcontainer.Name == "sdc" && sdcEnabled {
+			// Update MDM env variable for "sdc" container
+			for i, env := range initcontainer.Envs {
 				if env.Name == "MDM" {
-					env.Value = mdmVar
-					newmdm = env
-					k = c
+					initcontainer.Envs[i].Value = mdmVar
 					break
 				}
 			}
-			initenv[k] = newmdm
+		}
+		if initcontainer.Name != "sdc" || sdcEnabled {
 			newInitContainers = append(newInitContainers, initcontainer)
 		}
 	}
-	cr.Spec.Driver.InitContainers = newInitContainers
-	if len(cr.Spec.Driver.InitContainers) == 0 && sdcEnabled {
-		cr.Spec.Driver.InitContainers = []csmv1.ContainerTemplate{
-			{
-				Name: "sdc",
-				Envs: []corev1.EnvVar{
-					{
-						Name:  "MDM",
-						Value: mdmVar,
-					},
-				},
-			},
-		}
+
+	// If no init containers left and SDC is enabled, add a new one
+	if len(newInitContainers) == 0 && sdcEnabled {
+		newInitContainers = append(newInitContainers, csmv1.ContainerTemplate{
+			Name: "sdc",
+			Envs: []corev1.EnvVar{{Name: "MDM", Value: mdmVar}},
+		})
 	}
+	cr.Spec.Driver.InitContainers = newInitContainers
 
-	for _, sidecar := range cr.Spec.Driver.SideCars {
-		if sidecar.Name == "sdc-monitor" {
-			sidenv := sidecar.Envs
-			var updatenv corev1.EnvVar
-			j := 0
-			for c, env := range sidenv {
+	// Update sidecar containers
+	for i := range cr.Spec.Driver.SideCars {
+		if cr.Spec.Driver.SideCars[i].Name == "sdc-monitor" {
+			// Update MDM env variable for "sdc-monitor" sidecar container
+			for j, env := range cr.Spec.Driver.SideCars[i].Envs {
 				if env.Name == "MDM" {
-					env.Value = mdmVar
-					updatenv = env
-					j = c
+					cr.Spec.Driver.SideCars[i].Envs[j].Value = mdmVar
```

Comment:
If SDC is enabled, we are adding the MDM values from env to the CR here, are we not ? But one of the asks was to not have the MDM values show up in the CR. @rishabhatdell @anathoodell Doesn't this make the MDM value show up in the CSM resource ?